### PR TITLE
[CSFS-111]  Develop a Customer details list view page

### PR DESCRIPTION
**Pull request by rubyade**
Screenshot (60)

### CODE REVIEW

File: submissions/rubyAdeniregun/php/create_table.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    $pdo = new PDO("mysql:host=localhost;dbname=ecommercesite", "root", "mysql");
+    
+    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+
+} catch(PDOException $e){
+    die("ERROR: could not connect. " . $e->getmessage());
+}
+
+try{
+    $sql = "CREATE TABLE customer_details(
```
Comment (adaugochi) [inline submissions/rubyAdeniregun/php/create_table.php:15]: You are not expected to create a new table. You are to work with the e-commerce database  task you did previously

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Review by adaugochi — CHANGES_REQUESTED** [review]
Kindly read through the task again. You are not doing what the task required of you

File: submissions/rubyAdeniregun/php/listview.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+require_once("db_connection.php");
+
+echo "<h5> List View Page </h5>";
+
+try{
```
Comment (adaugochi) [inline submissions/rubyAdeniregun/php/listview.php:7]: This is what looks like what you are asked to do.

But, With the way you implement this, each time I reload this page, this query `"SELECT * FROM customer_details"` will always run. Which is not what we want to do. You have to look for a way to store the result gotten from this query in a cookie. So if cookie is set then you don't want to run the query again but get the data from the cookie store

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Review by adaugochi — CHANGES_REQUESTED** [review]
You haven't done what is required of you

File: submissions/rubyAdeniregun/php/load_cookie.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?php
```
Comment (adaugochi) [inline submissions/rubyAdeniregun/php/load_cookie.php:1]: This file isn't necessary. We ain't doing anything with session

File: submissions/rubyAdeniregun/php/listview.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    $sql= "SELECT * FROM customer_details";
+    $result= $pdo->query($sql);
+    if ($result->rowCount() > 0){
```
Comment (adaugochi) [inline submissions/rubyAdeniregun/php/listview.php:10]: Now there is progress

But if I reload this `listview.php` this query (`"SELECT * FROM customer_details"`) will always run. But, that is not what we want. We don't want to always run this query all the time, then store it in a cookie. 

Now. when we reload the `listview.php`, we want `$result` to be that coming from that stored in the cookie as long as the cookie is set. If cookie has expired, then `$result` should be that comming from the query

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by Ireti203** [comment]
@rubyade You still haven't made the changes @adaugochi requested in the listview.php file.

File: submissions/rubyAdeniregun/php/listview.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+require_once("db_connection.php");
+
+echo "<h5> List View Page </h5>";
+
+setcookie("name", "Customer Details", time()+3600, "/", "", false);
```
Comment (adaugochi) [inline submissions/rubyAdeniregun/php/listview.php:7]: This isn't doing anything

File: submissions/rubyAdeniregun/php/listview.php (hunk 4 above)
Comment (adaugochi) [reply submissions/rubyAdeniregun/php/listview.php:12]: The idea is to store the `$result` gotten from the query in a cookie. This action should be within a function.

Now. when we reload the `listview.php`, we want `$result` to be that coming from that stored in the cookie as long as the cookie is set. If the cookie has expired, then `$result` should be that coming from the query

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Review by rubyade — COMMENTED** [review]
Screenshot (60)

**Pull request merged**